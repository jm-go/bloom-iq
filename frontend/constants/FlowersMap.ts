export const FlowersMap = [
  { id: 0, name: 'Pink Primrose', image: require('@/assets/images/oxford-flowers/pink_primrose.jpg') },
  { id: 1, name: 'Hard-Leaved Pocket Orchid', image: require('@/assets/images/oxford-flowers/hard_leaved_pocket_orchid.jpg') },
  { id: 2, name: 'Canterbury Bells', image: require('@/assets/images/oxford-flowers/canterbury_bells.jpg') },
  { id: 3, name: 'Sweet Pea', image: require('@/assets/images/oxford-flowers/sweet_pea.jpg') },
  { id: 4, name: 'English Marigold', image: require('@/assets/images/oxford-flowers/english_marigold.jpg') },
  { id: 5, name: 'Tiger Lily', image: require('@/assets/images/oxford-flowers/tiger_lily.jpg') },
  { id: 6, name: 'Moon Orchid', image: require('@/assets/images/oxford-flowers/moon_orchid.jpg') },
  { id: 7, name: 'Bird of Paradise', image: require('@/assets/images/oxford-flowers/bird_of_paradise.jpg') },
  { id: 8, name: 'Monkshood', image: require('@/assets/images/oxford-flowers/monkshood.jpg') },
  { id: 9, name: 'Globe Thistle', image: require('@/assets/images/oxford-flowers/globe_thistle.jpg') },
  { id: 10, name: 'Snapdragon', image: require('@/assets/images/oxford-flowers/snapdragon.jpg') },
  { id: 11, name: "Colt's Foot", image: require('@/assets/images/oxford-flowers/colts_foot.jpg') },
  { id: 12, name: 'King Protea', image: require('@/assets/images/oxford-flowers/king_protea.jpg') },
  { id: 13, name: 'Spear Thistle', image: require('@/assets/images/oxford-flowers/spear_thistle.jpg') },
  { id: 14, name: 'Yellow Iris', image: require('@/assets/images/oxford-flowers/yellow_iris.jpg') },
  { id: 15, name: 'Globe-Flower', image: require('@/assets/images/oxford-flowers/globe_flower.jpg') },
  { id: 16, name: 'Purple Coneflower', image: require('@/assets/images/oxford-flowers/purple_coneflower.jpg') },
  { id: 17, name: 'Peruvian Lily', image: require('@/assets/images/oxford-flowers/peruvian_lily.jpg') },
  { id: 18, name: 'Balloon Flower', image: require('@/assets/images/oxford-flowers/balloon_flower.jpg') },
  { id: 19, name: 'Giant White Arum Lily', image: require('@/assets/images/oxford-flowers/giant_white_arum_lily.jpg') },
  { id: 20, name: 'Fire Lily', image: require('@/assets/images/oxford-flowers/fire_lily.jpg') },
  { id: 21, name: 'Pincushion Flower', image: require('@/assets/images/oxford-flowers/pincushion_flower.jpg') },
  { id: 22, name: 'Fritillary', image: require('@/assets/images/oxford-flowers/fritillary.jpg') },
  { id: 23, name: 'Red Ginger', image: require('@/assets/images/oxford-flowers/red_ginger.jpg') },
  { id: 24, name: 'Grape Hyacinth', image: require('@/assets/images/oxford-flowers/grape_hyacinth.jpg') },
  { id: 25, name: 'Corn Poppy', image: require('@/assets/images/oxford-flowers/corn_poppy.jpg') },
  { id: 26, name: 'Prince of Wales Feathers', image: require('@/assets/images/oxford-flowers/prince_of_wales_feathers.jpg') },
  { id: 27, name: 'Stemless Gentian', image: require('@/assets/images/oxford-flowers/stemless_gentian.jpg') },
  { id: 28, name: 'Artichoke', image: require('@/assets/images/oxford-flowers/artichoke.jpg') },
  { id: 29, name: 'Sweet William', image: require('@/assets/images/oxford-flowers/sweet_william.jpg') },
  { id: 30, name: 'Carnation', image: require('@/assets/images/oxford-flowers/carnation.jpg') },
  { id: 31, name: 'Garden Phlox', image: require('@/assets/images/oxford-flowers/garden_phlox.jpg') },
  { id: 32, name: 'Love in the Mist', image: require('@/assets/images/oxford-flowers/love_in_the_mist.jpg') },
  { id: 33, name: 'Mexican Aster', image: require('@/assets/images/oxford-flowers/mexican_aster.jpg') },
  { id: 34, name: 'Alpine Sea Holly', image: require('@/assets/images/oxford-flowers/alpine_sea_holly.jpg') },
  { id: 35, name: 'Ruby-Lipped Cattleya', image: require('@/assets/images/oxford-flowers/ruby_lipped_cattleya.jpg') },
  { id: 36, name: 'Cape Flower', image: require('@/assets/images/oxford-flowers/cape_flower.jpg') },
  { id: 37, name: 'Great Masterwort', image: require('@/assets/images/oxford-flowers/great_masterwort.jpg') },
  { id: 38, name: 'Siam Tulip', image: require('@/assets/images/oxford-flowers/siam_tulip.jpg') },
  { id: 39, name: 'Lenten Rose', image: require('@/assets/images/oxford-flowers/lenten_rose.jpg') },
  { id: 40, name: 'Barberton Daisy', image: require('@/assets/images/oxford-flowers/barberton_daisy.jpg') },
  { id: 41, name: 'Daffodil', image: require('@/assets/images/oxford-flowers/daffodil.jpg') },
  { id: 42, name: 'Sword Lily', image: require('@/assets/images/oxford-flowers/sword_lily.jpg') },
  { id: 43, name: 'Poinsettia', image: require('@/assets/images/oxford-flowers/poinsettia.jpg') },
  { id: 44, name: 'Bolero Deep Blue', image: require('@/assets/images/oxford-flowers/bolero_deep_blue.jpg') },
  { id: 45, name: 'Wallflower', image: require('@/assets/images/oxford-flowers/wallflower.jpg') },
  { id: 46, name: 'Marigold', image: require('@/assets/images/oxford-flowers/marigold.jpg') },
  { id: 47, name: 'Buttercup', image: require('@/assets/images/oxford-flowers/buttercup.jpg') },
  { id: 48, name: 'Oxeye Daisy', image: require('@/assets/images/oxford-flowers/oxeye_daisy.jpg') },
  { id: 49, name: 'Common Dandelion', image: require('@/assets/images/oxford-flowers/common_dandelion.jpg') },
  { id: 50, name: 'Petunia', image: require('@/assets/images/oxford-flowers/petunia.jpg') },
  { id: 51, name: 'Wild Pansy', image: require('@/assets/images/oxford-flowers/wild_pansy.jpg') },
  { id: 52, name: 'Primula', image: require('@/assets/images/oxford-flowers/primula.jpg') },
  { id: 53, name: 'Sunflower', image: require('@/assets/images/oxford-flowers/sunflower.jpg') },
  { id: 54, name: 'Pelargonium', image: require('@/assets/images/oxford-flowers/pelargonium.jpg') },
  { id: 55, name: 'Bishop of Llandaff', image: require('@/assets/images/oxford-flowers/bishop_of_llandaff.jpg') },
  { id: 56, name: 'Gaura', image: require('@/assets/images/oxford-flowers/gaura.jpg') },
  { id: 57, name: 'Geranium', image: require('@/assets/images/oxford-flowers/geranium.jpg') },
  { id: 58, name: 'Orange Dahlia', image: require('@/assets/images/oxford-flowers/orange_dahlia.jpg') },
  { id: 59, name: 'Pink-Yellow Dahlia', image: require('@/assets/images/oxford-flowers/pink_yellow_dahlia.jpg') },
  { id: 60, name: 'Cautleya Spicata', image: require('@/assets/images/oxford-flowers/cautleya_spicata.jpg') },
  { id: 61, name: 'Japanese Anemone', image: require('@/assets/images/oxford-flowers/japanese_anemone.jpg') },
  { id: 62, name: 'Black-Eyed Susan', image: require('@/assets/images/oxford-flowers/black_eyed_susan.jpg') },
  { id: 63, name: 'Silverbush', image: require('@/assets/images/oxford-flowers/silverbush.jpg') },
  { id: 64, name: 'Californian Poppy', image: require('@/assets/images/oxford-flowers/californian_poppy.jpg') },
  { id: 65, name: 'Osteospermum', image: require('@/assets/images/oxford-flowers/osteospermum.jpg') },
  { id: 66, name: 'Spring Crocus', image: require('@/assets/images/oxford-flowers/spring_crocus.jpg') },
  { id: 67, name: 'Bearded Iris', image: require('@/assets/images/oxford-flowers/bearded_iris.jpg') },
  { id: 68, name: 'Windflower', image: require('@/assets/images/oxford-flowers/windflower.jpg') },
  { id: 69, name: 'Tree Poppy', image: require('@/assets/images/oxford-flowers/tree_poppy.jpg') },
  { id: 70, name: 'Gazania', image: require('@/assets/images/oxford-flowers/gazania.jpg') },
  { id: 71, name: 'Azalea', image: require('@/assets/images/oxford-flowers/azalea.jpg') },
  { id: 72, name: 'Water Lily', image: require('@/assets/images/oxford-flowers/water-lily.jpg') },
  { id: 73, name: 'Rose', image: require('@/assets/images/oxford-flowers/rose.jpg') },
  { id: 74, name: 'Thorn Apple', image: require('@/assets/images/oxford-flowers/thorn-apple.jpg') },
  { id: 75, name: 'Morning Glory', image: require('@/assets/images/oxford-flowers/morning-glory.jpg') },
  { id: 76, name: 'Passion Flower', image: require('@/assets/images/oxford-flowers/passion-flower.jpg') },
  { id: 77, name: 'Lotus', image: require('@/assets/images/oxford-flowers/lotus.jpg') },
  { id: 78, name: 'Toad Lily', image: require('@/assets/images/oxford-flowers/toad-lily.jpg') },
  { id: 79, name: 'Anthurium', image: require('@/assets/images/oxford-flowers/anthurium.jpg') },
  { id: 80, name: 'Frangipani', image: require('@/assets/images/oxford-flowers/frangipani.jpg') },
  { id: 81, name: 'Clematis', image: require('@/assets/images/oxford-flowers/clematis.jpg') },
  { id: 82, name: 'Hibiscus', image: require('@/assets/images/oxford-flowers/hibiscus.jpg') },
  { id: 83, name: 'Columbine', image: require('@/assets/images/oxford-flowers/columbine.jpg') },
  { id: 84, name: 'Desert Rose', image: require('@/assets/images/oxford-flowers/desert_rose.jpg') },
  { id: 85, name: 'Tree Mallow', image: require('@/assets/images/oxford-flowers/tree_mallow.jpg') },
  { id: 86, name: 'Magnolia', image: require('@/assets/images/oxford-flowers/magnolia.jpg') },
  { id: 87, name: 'Cyclamen', image: require('@/assets/images/oxford-flowers/cyclamen.jpg') },
  { id: 88, name: 'Watercress', image: require('@/assets/images/oxford-flowers/watercress.jpg') },
  { id: 89, name: 'Canna Lily', image: require('@/assets/images/oxford-flowers/canna_lily.jpg') },
  { id: 90, name: 'Hippeastrum', image: require('@/assets/images/oxford-flowers/hippeastrum.jpg') },
  { id: 91, name: 'Bee Balm', image: require('@/assets/images/oxford-flowers/bee_balm.jpg') },
  { id: 92, name: 'Ball Moss', image: require('@/assets/images/oxford-flowers/ball_moss.jpg') },
  { id: 93, name: 'Foxglove', image: require('@/assets/images/oxford-flowers/foxglove.jpg') },
  { id: 94, name: 'Bougainvillea', image: require('@/assets/images/oxford-flowers/bougainvillea.jpg') },
  { id: 95, name: 'Camellia', image: require('@/assets/images/oxford-flowers/camellia.jpg') },
  { id: 96, name: 'Mallow', image: require('@/assets/images/oxford-flowers/mallow.jpg') },
  { id: 97, name: 'Mexican Petunia', image: require('@/assets/images/oxford-flowers/mexican_petunia.jpg') },
  { id: 98, name: 'Bromelia', image: require('@/assets/images/oxford-flowers/bromelia.jpg') },
  { id: 99, name: 'Blanket Flower', image: require('@/assets/images/oxford-flowers/blanket_flower.jpg') },
  { id: 100, name: 'Trumpet Creeper', image: require('@/assets/images/oxford-flowers/trumpet_creeper.jpg') },
  { id: 101, name: 'Blackberry Lily', image: require('@/assets/images/oxford-flowers/blackberry_lily.jpg') },
];

  