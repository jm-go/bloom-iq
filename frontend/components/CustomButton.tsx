import React, { FC } from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';
import { AntDesign } from '@expo/vector-icons';
import { Colors } from '@/constants/Colors';

interface ButtonProps {
  text: string;
  onPress: () => void;
  backgroundColor?: string;
  icon?: keyof typeof AntDesign.glyphMap;
  textColor?: string;
  width?: number;
}

const CustomButton: FC<ButtonProps> = ({
  text,
  onPress,
  backgroundColor = Colors.dark.primaryButton,
  icon,
  textColor = Colors.dark.text,
  width,
}) => {
  return (
    <TouchableOpacity
    style={[
      styles.button,
      {
        backgroundColor,
        width: width ?? undefined,
        minWidth: width ? undefined : 150,
        paddingHorizontal: width ? 0 : 20,
      },
    ]}
    onPress={onPress}
  >
      {icon && <AntDesign name={icon} size={24} color={textColor} />}
      <Text style={[styles.text, { color: textColor }]}>{text}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    padding: 15,
    borderRadius: 50,
    height: 60,
  },
  text: {
    fontSize: 16,
    marginLeft: 8,
  },
});

export default CustomButton;
