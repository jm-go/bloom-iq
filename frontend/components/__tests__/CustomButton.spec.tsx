import React from 'react';
import renderer from 'react-test-renderer';
import CustomButton from '../CustomButton';
import { Colors } from '@/constants/Colors';

describe('test CustomButton.tsx', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <CustomButton text="Take a Photo" onPress={() => {}} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with a custom background color and icon', () => {
    const tree = renderer.create(
      <CustomButton
        text="Retake"
        onPress={() => {}}
        backgroundColor={Colors.dark.primaryButton}
        icon="reload1"
        width={180}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with a different text color', () => {
    const tree = renderer.create(
      <CustomButton
        text="Upload"
        onPress={() => {}}
        backgroundColor={Colors.dark.secondaryButton}
        textColor={Colors.dark.text}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
