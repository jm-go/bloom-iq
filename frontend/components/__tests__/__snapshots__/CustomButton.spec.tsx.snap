// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CustomButton.tsx renders correctly with a custom background color and icon 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#6666ff",
      "borderRadius": 50,
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "center",
      "minWidth": undefined,
      "opacity": 1,
      "padding": 15,
      "paddingHorizontal": 0,
      "width": 180,
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#ECEDEE",
          "fontSize": 24,
        },
        undefined,
        {
          "fontFamily": "anticon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    î˜–
  </Text>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "marginLeft": 8,
        },
        {
          "color": "#ECEDEE",
        },
      ]
    }
  >
    Retake
  </Text>
</View>
`;

exports[`test CustomButton.tsx renders correctly with a different text color 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#755dd3",
      "borderRadius": 50,
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "center",
      "minWidth": 150,
      "opacity": 1,
      "padding": 15,
      "paddingHorizontal": 20,
      "width": undefined,
    }
  }
>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "marginLeft": 8,
        },
        {
          "color": "#ECEDEE",
        },
      ]
    }
  >
    Upload
  </Text>
</View>
`;

exports[`test CustomButton.tsx renders correctly with default props 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#6666ff",
      "borderRadius": 50,
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "center",
      "minWidth": 150,
      "opacity": 1,
      "padding": 15,
      "paddingHorizontal": 20,
      "width": undefined,
    }
  }
>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "marginLeft": 8,
        },
        {
          "color": "#ECEDEE",
        },
      ]
    }
  >
    Take a Photo
  </Text>
</View>
`;
